apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableClientPartial.cs

```diff
@@ -46,19 +48,45 @@ public abstract partial class BigtableClient
         /// "default" application profile will be used.
         /// </param>
         /// <returns>The task representing the created <see cref="BigtableClient"/>.</returns>
-        public static async Task<BigtableClient> CreateAsync(ServiceEndpoint endpoint = null, BigtableServiceApiSettings settings = null, string appProfileId = null)
+        public static async Task<BigtableClient> CreateAsync(BigtableServiceApiSettings settings = null, ClientCreationSettings clientCreationSettings = null, string appProfileId = null)
         {
-            var client = await BigtableServiceApiClient.CreateAsync(endpoint, settings).ConfigureAwait(false);
-            return new BigtableClientImpl(client, appProfileId) { UnderlyingClientSettings = settings };
+            clientCreationSettings?.Validate();
+            var channelCredentials = clientCreationSettings?.Credentials;
+            // Use default credentials if none given.
+            if (channelCredentials == null)
+            {
+                var credentials = await GoogleCredential.GetApplicationDefaultAsync().ConfigureAwait(false);
+                if (credentials.IsCreateScopedRequired)
+                {
+                    credentials = credentials.CreateScoped(BigtableServiceApiClient.DefaultScopes);
+                }
+                channelCredentials = credentials.ToChannelCredentials();
+            }
+            var clientCount = clientCreationSettings?.ClientCount ?? Environment.ProcessorCount;
+
+            var endpoint = clientCreationSettings?.ServiceEndpoint ?? BigtableServiceApiClient.DefaultEndpoint;
+            var clients = new BigtableServiceApiClient[clientCount];
+            var shutdowns = new Func<Task>[clientCount];
+            // Fill clients[] with BigtableServiceApiClient instances, each with specific channel
+            for (int i = 0; i < clientCount; i++)
+            {
+                var channel = new Channel(endpoint.Host, endpoint.Port, channelCredentials);
+                clients[i] = BigtableServiceApiClient.Create(channel, clientCreationSettings?.BigtableServiceApiSettings);
+                shutdowns[i] = channel.ShutdownAsync;
+            }
+            Func<Task> shutdown = () => Task.WhenAll(shutdowns.Select(x => x()));
+
+            return new BigtableClientImpl(appProfileId, clients) { UnderlyingClientSettings = settings };
```

Comment:
`settings` here should be `clientCreationSettings?.BigtableServiceApiSettings`. Actually, I would leave this as-is, and capture `clientCreationSettings?.BigtableServiceApiSettings` in a local above the for loop, then use that local inside the loop as well.